momepy/preprocessing.py

```diff
@@ -740,3 +745,368 @@ def _get_extrapolated_line(coords, tolerance, point=False):
     if point:
         return b
     return pygeos.linestrings([a, b])
+
+
+def consolidate_intersections(
+    graph,
+    tolerance=30,
+    rebuild_graph=True,
+    rebuild_edges_method="spider",
+    directed=None,
+    x_col="x",
+    y_col="y",
+    edge_from_col="from",
+    edge_to_col="to",
+):
+    """
+    Consolidate close street intersections into a single node, collapsing short edges.
+
+    If rebuild_graph is True, new edges are drawn according to rebuild_edges_method which is one of:
+
+    1. Linear reconstruction:
+        Edges are linearly extended from original endpoints until the new nodes. This method preserves
+        most faithfully the network geometry.
+    2. Spider-web reconstruction:
+        Edges are cropped within a buffer of the new endpoints and linearly extended from there. This
+        method improves upon linear reconstruction by mantaining, when possible, network planarity.
+    3. As-the-crow-flies reconstruction:
+        Edges are ignored and new edges are built as straightlines between new origin and new
+        destination. This method ignores geometry, but efficiently preserves adjacency.
+
+    If rebuild_graph is False, graph is returned with consolidated nodes but without reconstructed
+    edges i.e. graph is intentionally disconnected.
+
+    Graph must be configured so that
+
+    1. All nodes have attributes determining their x and y coordinates;
+    2. All edges have attributes determining their origin, destination, and geometry;
+    3. Graph itself has an attribute determining its crs.
+
+    Parameters
+    ----------
+    graph : Networkx.MultiGraph or Networkx.MultiDiGraph
+
+    tolerance : float
+        distance in network units below which nodes will be consolidated
+
+    rebuild_graph: Boolean
+
+    rebuild_edges_method: string
+        'spider' or 'linear' or 'crow_flies'
```

Comment:
I am not sure if these options are clear based on their names. `spider` is okay but as crow flies is linear line... What about using `extend` instead of `linear` and `euclidean` instead of `crow_flies`?